### DISCUSSION TITLE
No Solidity 8 versions of LinkToken.sol or StandardToken.sol

### DISCUSSION THREAD

**Discussion by JohnEMagee** [smartcontractkit/full-blockchain-solidity-course-py#1292] [Q&A]
So unsurprisingly next mock contract runs into another problem.  This one doesn't seem solvable for linktoken/erc677token/standard token as there are no test versions readily available for solidity 0.8 of these .sol files.

At this point, I'm 100% ready to give up this tutorial and going to drop a review about how this does no work for mac m1s as no one seems to be making ANY effort to make this work.  I spend half my time trying to fix compatibilities.  solidity is at 0.8 and this tutorial works in 0.6, and because of m1 incompatibilities that no one seems to admit exist it's almost impossible to get through this...i wanted to learn how to make NFTs but at this point i can't even finish the lottery.  It gets more frustrating every day and there's very little help provided on this git hub, the recorder of the video seems absent as does FCC.  Please advise me when any SERIOUS attention is paid attention to the numerous Mac M1 issues because at this point it's pointless to have to go through this every day, i'm not learning anything really

**Comment by cromewar** [comment]
Hello @JohnEMagee 
I understand your frustration on this, but I'm sorry to say there is not much we can do about it. M1 Macs are really new on the space with a complete new architecture, the amount un M1 users which also try to develop smart contracts are actually really small, so is hard for developer to focus all their attention on small niches, as same as game developers does not develop much games for linux, or adobe does no port the adobe suit to it. The official Solidity developers said they are working on a full compatible version of solidity as you can see [here](https://github.com/ethereum/solidity/issues/12291) but we don't know when.

Take in mind that the only way for any user to help you is to be able to reproduce the error on their own machines, most of them will not have any way to do it. most users are windows ones, and the rest are Mac (not m1 chip) and a small amount of linux users. So of course it'll hard to find "Seriuos" attention on this matter due to the limitations I've described above.

This is a common problem on cutting edge technology is almost impossible to provide support or solutions for very specific niches, anyways the technology will grow and mature with time and also the frameworks and technologies related. So I just can say, be a little more patient with this and on the meantime you can try other things. I almost sure the NFT lesson will not give you the same problems with old dependencies.

Cheers.

**Comment by n4n0b1t3** [comment]
@JohnEMagee Hey buddy, we're all in the same boat here. Let me remind you that you're struggling through a free course on YouTube right now, and you didn't buy a product on Amazon that you can return and give a bad review if you don't like it. I have my frustrating moments too. Lesson 7 is long and complicated. I had also just started writing everything using the latest Solidity version, but I've found that I run into too many problems and just don't get anywhere. You have to remember that the ability to use different versions is a really good feature, and it improves a developer's ability to learn how to use them. 
99% of the people on this forum are on the same path as you and are helping others solve problems. Judging by your posts you are doing a great job and even helping others, **I really don't want to lose you**. 

When I started learning web development in 1996, we only had (fricking outdated) books. Since these were sold on shelves in real bookstores, it was important for marketing reasons that the book had a very large spine. At that time there were editors whose only job was to artificially inflate the content of books so that they became thick. A book at that time had the price of $60 USD converted to today's money. Most of the time only 30% of these books were usable, the rest consisted of repetitions and there were soooo many errors.

Be glad you have a new MacBook with an M1 core and that you can use awesome free courses and talk about them in forums like these. As far as web development goes, it's never been better. By the way, before this course I was learning from the best-rated Udemy course and it was far more outdated and with no discussion group or updated repository on GitHub.

**Reply by n4n0b1t3** [reply]
> Tell me how this does anything to help me? Toxic positivity condescension is not helpful.

1. This will be on my next t-shirt!
2. suggested read https://www.goodreads.com/book/show/27213329-grit
3. We are not here to help you!
